drivers/modem/modem_cmd_handler.c

```diff
@@ -476,7 +476,15 @@ static int _modem_cmd_send(struct modem_iface *iface,
 	struct modem_cmd_handler_data *data;
 	int ret;
 
-	if (!iface || !handler || !handler->cmd_handler_data || !buf) {
+	if (!iface || !handler || !handler->cmd_handler_data || !buf || !sem) {
+		return -EINVAL;
+	}
+
+	if (K_TIMEOUT_EQ(timeout, K_NO_WAIT)) {
+		/* semaphore is not needed if there is no timeout */
+		sem = NULL;
+	} else if (!sem) {
+		/* cannot respect timeout without semaphore */
 		return -EINVAL;
 	}
```

Comment:
The first evaluation of the semaphore (ln 479) always fails if timeout is used when no sem is sent.

Reworking the line to:
` if (!iface || !handler || !handler->cmd_handler_data || !buf )`


Billy...

Reply:
The UBlox sara-modem now fails

Reply:
@WilliamGFish You are right, `!sem` shouldn't be there. I left it there by mistake... Will you send a PR?

Reply:
@WilliamGFish I do agree. This is introducing an error when called without sem nor timeout [here](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/modem/ublox-sara-r4.c#L365).

Reply:
#30458